plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -416,7 +416,7 @@ public IdBasedFieldMapper(Map<Integer, OrcColumn> idToColumnMappingForFile, Map<
         @Override
         public OrcColumn get(String fieldName)
         {
-            int fieldId = nameToIdMappingForTableColumns.get(fieldName);
+            int fieldId = requireNonNull(nameToIdMappingForTableColumns.get(fieldName), format("Id mapping for field %s is null", fieldName));
```

Comment:
use `requireNonNull` that takes a lambda to construct message lazily 

also "mapping is null" -> "no mapping for field "